### PR TITLE
[LoongArch] Emit R_LARCH_RELAX when expanding some LoadAddress

### PR DESCRIPTION
**Pull request by MQ-mengqing**
Emit relax relocs when expand non-large la.pcrel and non-large la.got on llvm-mc stage, which like what does on GAS.
1, la.pcrel -> PCALA_HI20 + RELAX + PCALA_LO12 + RELAX
2, la.got -> GOT_PC_HI20 + RELAX + GOT_PC_LO12 + RELAX

### CODE REVIEW

**Comment by MQ-mengqing** [comment]
Add: @SixWeining @wangleiat @xry111 @xen0n @MaskRay .

File: llvm/test/MC/LoongArch/Relocations/relax-la.s [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+# CHECK-NEXT:      0x8 R_LARCH_GOT_PC_HI20 .L1 0x0
+# CHECK-NEXT:      0x8 R_LARCH_RELAX - 0x0
+# CHECK-NEXT:      0xC R_LARCH_GOT_PC_LO12 .L1 0x0
+# CHECK-NEXT:      0xC R_LARCH_RELAX - 0x0
+# CHECK-NEXT:    }
+
+.text
+.L1:
+  la.pcrel $a0, .L1
+  la.got   $a0, .L1
```
Comment (MaskRay) [inline llvm/test/MC/LoongArch/Relocations/relax-la.s:19]: `llvm/test/MC/LoongArch/Macros/macros-la.s` can be enhanced to test relocations.

**Comment by MaskRay** [comment]
> Emit relax relocs when expand some LoadAddress

I suggest `[LoongArch] Emit R_LARCH_RELAX when expanding some LoadAddress`

File: llvm/test/MC/LoongArch/Macros/macros-la.s [hunk 2]
```diff
@@ -1,66 +1,110 @@
...
 # CHECK-NEXT: lu52i.d $a0, $a0, %abs64_hi12(sym_abs)
+# RELOC:      R_LARCH_ABS_HI20 sym_abs 0x0
+# RELOC-NEXT: R_LARCH_ABS_LO12 sym_abs 0x0
+# RELOC-NEXT: R_LARCH_ABS64_LO20 sym_abs 0x0
+# RELOC-NEXT: R_LARCH_ABS64_HI12 sym_abs 0x0
 
 la.pcrel $a0, sym_pcrel
 # CHECK:      pcalau12i $a0, %pc_hi20(sym_pcrel)
 # CHECK-NEXT: addi.d $a0, $a0, %pc_lo12(sym_pcrel)
+# RELOC:      R_LARCH_PCALA_HI20 sym_pcrel 0x0
```
Comment (SixWeining) [inline llvm/test/MC/LoongArch/Macros/macros-la.s:20]: Had better to use `RELOC-NEXT` for more accurate.

Seems we also should change `CHECK` to `CHECK-NEXT` before this PR.

File: llvm/test/MC/LoongArch/Macros/macros-la.s [hunk 3]
```diff
@@ -1,66 +1,110 @@
...
+# RUN: llvm-readobj -r %t | FileCheck %s --check-prefix=RELOC
+# RUN: llvm-mc --filetype=obj --triple=loongarch64 --mattr=+relax %s -o %t.relax
+# RUN: llvm-readobj -r %t.relax | FileCheck %s --check-prefixes=RELOC,RELAX
 
 la.abs $a0, sym_abs
 # CHECK:      lu12i.w $a0, %abs_hi20(sym_abs)
 # CHECK-NEXT: ori $a0, $a0, %abs_lo12(sym_abs)
 # CHECK-NEXT: lu32i.d $a0, %abs64_lo20(sym_abs)
 # CHECK-NEXT: lu52i.d $a0, $a0, %abs64_hi12(sym_abs)
+# RELOC:      R_LARCH_ABS_HI20 sym_abs 0x0
```
Comment (SixWeining) [inline llvm/test/MC/LoongArch/Macros/macros-la.s:12]: It's better to add start/end flags like: 
```
# RELOC:          Relocations [
# RELOC-NEXT:       Section ({{.*}}) .rela.text {

... your check goes here

# RELOC-NEXT:       }
# RELOC-NEXT:     ]
```

**Review by SixWeining — APPROVED** [review]
LGTM

**Pull request merged**